src/kinetics/KineticsFactory.cpp

```diff
@@ -137,8 +143,14 @@ shared_ptr<Kinetics> newKinetics(const vector<shared_ptr<ThermoPhase>>& phases,
                                  const string& filename,
                                  const string& phase_name)
 {
+    if (phase_name != "" && phase_name != phases.at(0)->name()) {
+        warn_deprecated("newKinetics", "The reacting phase must be the first phase in "
+            "the 'phases' vector. Remove the 'phase_name' argument to use the first "
+            " phase by default. This warning will be an error after Cantera 3.0. ");
```

Comment:
Isn’t the parameter removed after Cantera 3.0?

Reply:
The `phase_name` parameter is gone, but having the reacting phase not in first position will raise an exception from `Kinetics::addThermo`.

Reply:
OK, I think I was making this part too complicated -- the `phase_name` parameter can be deprecated immediately. Specifying the first phase's name to `newKinetics` can issue a warning, and having the reacting phase in a different order will issue a warning from `Kinetics::addThermo`.